mqtt/src/main/scala/akka/stream/alpakka/mqtt/Mqtt.scala

```diff
@@ -79,26 +81,64 @@ final case class MqttConnectionSettings(
     auth: Option[(String, String)] = None,
     socketFactory: Option[SSLSocketFactory] = None,
     cleanSession: Boolean = true,
-    will: Option[MqttMessage] = None
+    will: Option[MqttMessage] = None,
+    automaticReconnect: Boolean = false,
+    keepAliveInterval: FiniteDuration = 60.seconds,
+    connectionTimeout: FiniteDuration = 30.seconds,
+    maxInFlight: Int = 10,
+    mqttVersion: Int = MqttConnectOptions.MQTT_VERSION_3_1_1,
+    serverUris: Option[Array[String]] = None,
+    sslHostnameVerifier: Option[HostnameVerifier] = None,
+    sslProperties: Option[Properties] = None
```

Comment:
Can you propose a way using a Scala Map? Properties must be one of the worst JDK classes I know.